regcore/db/django_models.py

```diff
@@ -40,9 +40,16 @@ def _transform(self, reg, version):
 
     def bulk_put(self, regs, version, root_label):
         """Store all reg objects"""
-        # This does not handle subparts. Ignoring that for now
-        Regulation.objects.filter(version=version,
-                                  label_string__startswith=root_label).delete()
+        # Delete any existing regulation objects for this version and
+        # root label.
+        # XXX: This does not handle subparts. Ignoring that for now
+        existing = Regulation.objects.filter(version=version,
+                                  label_string__startswith=root_label)
+        # It's possible for this queryset to be so large as to timeout
+        # the MySQL connection if executed in bulk. So this is ugly, but
+        # necessarily so.
+        map(lambda e: e.delete(), existing)
```

Comment:
:confused:  There's got to be a better solution. Pulling every entry from the db is going to be deadly slow; it might work when you're running with the data locally, but I'd expect any HTTP connection would time out?

Two thoughts -- have you looked at the generated query? It's possible (though unlikely) that django's not issuing a sensical query. Second, assuming the query itself is super slow, what do you think about removing the `ilike` and replacing it with an indexed `cfr_part` field? That should speed this up a bit -- even more if an index is added to the pair `(version, cfr_part)`. It'd also resolve the potential collision when one cfr_part is the prefix for another (e.g. `100` and `1005`).